### ISSUE TITLE
pillar graphics (toyland), but also other

### ISSUE THREAD

**Issue by DorpsGek** [OpenTTD/OpenTTD#4146]
**_LordAro_ opened the ticket and wrote:**

> see pic
> 
> doesn't look very good does it?

**Attachments**
- [bridge graphics problem.png](https://bugs.openttd.org/task/4146/getfile/6628/bridge+graphics+problem.png) (40.00 KiB)

**Reported version**: _trunk_
**Operating system**: _All_

---

**Comment by DorpsGek** [comment]
**_Rubidium_ wrote:**

> Looks fine to me, but then I don't know what I should be looking for as you didn't tell that...

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8846_

**Comment by DorpsGek** [comment]
**_frosch_ wrote:**

> I think this is about the length of bridgepillars. The pillarsprites include the pillars for both ends of the tile, so on sloped tiles one side is always too short.
> 
> This sprite-issue can be worked around using some "SubSprite"-magic like for halftiles.
> 
> IIRC there already is some FS feature request about this, but as I cannot find it, I keep this one.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8849_

**Comment by DorpsGek** [comment]
**_uni657_ wrote:**

> What a coincidence, right now I'm working on patch that fixes bridge pillars. It's almost done, only OpenGFX and NewGRF issues remain.
> http://www.tt-forums.net/viewtopic.php?f=33&t=50306&start=0

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8853_

**Comment by DorpsGek** [comment]
**_LordAro_ wrote:**

> @Rubidium Re: Waiting on Reporter:
> Just the fact the the extended supports are too small and out of alignment...

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8862_

**Comment by DorpsGek** [comment]
**_uni657_ wrote:**

> Change to v2: removed flag draw full back pillar
> 
> Stage 1 fixes vertically misaligned pillars of the bridges.
> Each diff applies only on top of the previous stage.
> 
> Sorry, "bridge_pillars_fix_stage3_v2 (tubular fix).diff" was damaged, added correct file.

**Attachments**
- [bridge_pillars_fix_stage1_v1.diff](https://bugs.openttd.org/task/4146/getfile/6646/bridge_pillars_fix_stage1_v1.diff) (8.40 KiB)
- [bridge_pillars_fix_stage1_v2.diff](https://bugs.openttd.org/task/4146/getfile/6649/bridge_pillars_fix_stage1_v2.diff) (8.54 KiB)
- [bridge_pillars_fix_stage2_v2 (cantilever fix).diff](https://bugs.openttd.org/task/4146/getfile/6650/bridge_pillars_fix_stage2_v2+%28cantilever+fix%29.diff) (1.17 KiB)
- [bridge_pillars_fix_stage3_v2 (tubular fix).diff](https://bugs.openttd.org/task/4146/getfile/6651/bridge_pillars_fix_stage3_v2+%28tubular+fix%29.diff) (1.43 KiB)
- [bridge_pillars_fix_stage4_v2 (detect opengfx newgrf).diff](https://bugs.openttd.org/task/4146/getfile/6652/bridge_pillars_fix_stage4_v2+%28detect+opengfx+newgrf%29.diff) (3.65 KiB)
- [bridge_pillars_fix_stage5_v2 (flag draw full back pillar).diff](https://bugs.openttd.org/task/4146/getfile/6653/bridge_pillars_fix_stage5_v2+%28flag+draw+full+back+pillar%29.diff) (0.84 KiB)
- [bridge_pillars_fix_stage3_v2 (tubular fix) correct file.diff](https://bugs.openttd.org/task/4146/getfile/6654/bridge_pillars_fix_stage3_v2+%28tubular+fix%29+correct+file.diff) (1.29 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8868_

**Comment by DorpsGek** [comment]
**_frosch_ wrote:**

> Why do the patches test for specific bridgetypes?
> What is wrong with just drawing the pillars the same for all bridgetypes including NewGRF bridges?

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8916_

**Comment by DorpsGek** [comment]
**_uni657_ wrote:**

> Sorry, I have removed all unnecessary tests. Remains only test for tubular and cantilever bridgetypes required to fix TTD bridges.

**Attachments**
- [bpf_stage1_v3.diff](https://bugs.openttd.org/task/4146/getfile/6690/bpf_stage1_v3.diff) (5.89 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8918_

**Comment by DorpsGek** [comment]
**_frosch_ wrote:**

> Usual bridges (read: all except original cantilever and tubular) now work in trunk due to uni637's patches :)
> 
> The issues of original cantilever and tubular bridges are caused by the graphics.
> I guess tubular could be fixed by adding some replacements to openttd.grf (like for the other grf bugs).
> For the cantilever bridge though that might not work so well.
> 

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8932_

**Comment by DorpsGek** [comment]
**_uni657_ wrote:**

> Small note about included patch, there are 2 tests for "if (IsInvisibilitySet(TO_BRIDGES)) return;", first in DrawBridgeMiddle() (line 1312 in tunnelbridge_cmd.cpp), just before drawing pillars for small/high bridges, second in DrawBridgePillars() (line 877 also in tunnelbridge_cmd.cpp), which is newer triggered. IMO the second could be removed.
> 
> It appears that half-pillar subsprites cut some pillars in New bridges 0.41 NewGRF (picture attached), so I slightly modified offset of the subsprites. TTD and OpenGFX pillars are unaffected by the change. Sorry, that I didn't check that earlier.
> 
> Added replacement sprites for tubular and cantilever bridges (including toyland climate).
> 
> There is one more thing, that is slightly out of scope of this patch. To add flag for NewGRF bridges to prevent skipping top two parts for back pillars, which are hidden by the bridge, as some NewGRF bridges are more transparent than the default TTD bridges. And therefore change drawfarpillar parameter to flags for DrawBridgePillars().

**Attachments**
- [new bridges 0.41.png](https://bugs.openttd.org/task/4146/getfile/6702/new+bridges+0.41.png) (15.87 KiB)
- [new bridges 0.41 compatible.diff](https://bugs.openttd.org/task/4146/getfile/6703/new+bridges+0.41+compatible.diff) (0.71 KiB)
- [tubular_fix.NFO](https://bugs.openttd.org/task/4146/getfile/6704/tubular_fix.NFO) (0.77 KiB)
- [fix_graphics_pillar_tubular2.pcx](https://bugs.openttd.org/task/4146/getfile/6705/fix_graphics_pillar_tubular2.pcx) (1.91 KiB)
- [cantilever_fix.NFO](https://bugs.openttd.org/task/4146/getfile/6706/cantilever_fix.NFO) (1.35 KiB)
- [fix_graphics_pillar_cantilever2.pcx](https://bugs.openttd.org/task/4146/getfile/6707/fix_graphics_pillar_cantilever2.pcx) (18.03 KiB)
- [change drawfarpillar to flags.diff](https://bugs.openttd.org/task/4146/getfile/6708/change+drawfarpillar+to+flags.diff) (2.37 KiB)
- [add draw full back pillar flag.diff](https://bugs.openttd.org/task/4146/getfile/6709/add+draw+full+back+pillar+flag.diff) (0.81 KiB)

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8934_

**Comment by DorpsGek** [comment]
**_uni657_ wrote:**

> Sorry for bugging, but will replacement sprites for tubular and cantilever bridges be included in trunk? They are looking the same as in the TTD. Also fix r20950 (which fixes the mentioned sprites) lacks sprites for the toyland climate (although no one plays it).

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment8972_

**Comment by DorpsGek** [comment]
**_frosch_ wrote:**

> Independent of the pillar sprites, the toyland stuff needs fixing.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146#comment9112_

**Comment by DorpsGek** [comment]
**_Rubidium_ closed the ticket.**

**Reason for closing: _Fixed_**

> In r21251 (toyland graphics). Our opinion is that your updated graphics look worse in game. We do not need to copy Transport Tycoon Deluxe's behaviour exactly.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4146_
